src/main/java/dev/jbang/source/ScriptSource.java

```diff
@@ -86,6 +92,22 @@ private ScriptSource(ResourceRef resourceRef, String content) {
 		this.script = content;
 	}
 
+	public List<String> getCompileOptions() {
+		return collectOptions("JAVAC_OPTIONS");
+	}
+
+	public String getCompilerBinary(String requestedJavaVersion) {
+		return resolveInJavaHome("javac", requestedJavaVersion);
+	}
+
+	public Predicate<ClassInfo> getMainFinder() {
+		return pubClass -> pubClass.method("main", BaseBuildCommand.STRINGARRAYTYPE) != null;
+	}
+
+	public String getExtension() {
```

Comment:
this wouldn't not always be true - i.e. `.jsh` for java shell...so maybe change to be a "deriveMainClassName" or similar ?

Reply:
vs `getMainFinder()`  ?  i'm not sure i'm following.  But if it's that, i'm not married to that name.  i just need to call it something so I could refactor and clean up.  :)

Reply:
its just a sligthly mismatch to have it on the ScriptSource as it represents something that can have varied extensions. They make more sense on a separate "BuildTool" interface/class.